protocols/netty/src/main/java/org/apache/james/protocols/netty/NettyServer.java

```diff
@@ -88,6 +88,8 @@ private NettyServer(Protocol protocol, Encryption secure, boolean proxyRequired,
         this.secure = secure;
         this.proxyRequired = proxyRequired;
         this.frameHandlerFactory = frameHandlerFactory;
+
+        this.setGracefulShutdown(false);
```

Comment:
Yes sorry to piggy back checks here, it turns related tests from 2.1s to 0.1s... Tests that I ended up running because of the EHLO changes...

Reply:
sounds great I'm all for gracefull shutdowns in general especially if they make the tests faster :D

Reply:
No here it is actually remove graceful shutdown to make tests faster

Reply:
shouldn't it be a parameter or an item in the factory then ? because this specific change is in production code not test code and effectively seems to  disable graceful shutdown for production servers 
I'm not very familiar with this layer of the code though I have dabbled in it a bit recently

Reply:
It seems the NettyServer is only used by test classes though, from what I can see in the code.

However I agree it would be more clean to add the option in the factory instead of forcing it in the constructor

Reply:
> because this specific change is in production code 

NettyServer is actually a testing server... That's why I did take the liberty. I did not relocate it in a test package in order not to change maven dependencies but this choice may be challenged.

Production code can be governed via a simple XML setting no worry.